source/Engine.cpp

```diff
@@ -1633,6 +1639,36 @@ void Engine::MoveShip(const shared_ptr<Ship> &ship)
 
 
 
+// Changing the players flagship during flight requires an update of all active
+// data that references the players flagship.
+void Engine::ChangePlayerFlagship(const shared_ptr<Ship> oldFlagship, shared_ptr<Ship> &newFlagship)
+{
+	// Remove active data in the old flagship.
+	if(oldFlagship)
+	{
+		oldFlagship->SetTargetAsteroid(nullptr);
+		oldFlagship->SetTargetFlotsam(nullptr);
+		oldFlagship->SetTargetShip(nullptr);
+		oldFlagship->SetTargetStellar(nullptr);
+		oldFlagship->SetTargetSystem(nullptr);
```

Comment:
`Ship::WasCaptured` does the same thing as here, so I think this can be moved in a separate function like `Ship::ClearTargets`.